clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java

```diff
@@ -158,6 +161,39 @@ public void testCloseAll() {
         }
     }
 
+    @Test
+    public void testReadFully() throws IOException {
+        RandomAccessFile raf = new RandomAccessFile(TestUtils.tempFile(), "rw");
```

Comment:
Why is it the case that the mocked file channel does no operation to the buffer? If you call `readFully` and you have a mocked `FileChannel.read` method, it can (and probably should) add bytes to the buffer. That way you don't need to mock the byte buffer and the test is more realistic.